src/deepsparse/loggers/helpers.py

```diff
@@ -50,8 +50,52 @@ def possibly_extract_value(value: Any, remainder: Optional[str] = None) -> Any:
     if not remainder:
         return value
 
-    for sub_remainders in remainder.split("."):
-        value = value.__getattribute__(sub_remainders)
+    # splits remainder into separate strings.
+    # each string can access the new nesting depth and optionally
+    # do indexing and slicing at this depth
+    for sub_remainder in remainder.split("."):
+        square_brackets = re.search(r"\[(.*?)\]", sub_remainder)
+        if square_brackets:
+            # retrieve the string without the square brackets
+            sub_remainder = sub_remainder.split("[")[0]
+        # access the new nesting depth
+        value = value.__getattribute__(sub_remainder)
+
+    if square_brackets:
+        return do_slicing_and_indexing(
+            value=value, square_brackets=square_brackets.group(1)
+        )
+
+    return value
+
+
+def do_slicing_and_indexing(
```

Comment:
non blocking comment, but this function seems super easily unit testable, so maybe add some tests for this one